docker/README.md

```diff
@@ -0,0 +1,131 @@
+## Running `teosd` in a docker container
+A `teos` image can be built from the Dockerfile located in `docker`. You can create the image by running:
+
+	cd rust-teos
+	docker build -f docker/Dockerfile -t teos .
+	
+Then we can create a container by running:
+
+	docker run -it teos
+
+To feed `teos` custom config options, there are two ways to do so: 
+
+1) Environment variables:
+
+Create a container by running:
+
+	docker run -it -e <ENV_VARIABLES> teos
+	
+Notice that the ENV variables are optional, if unset the corresponding default setting is used. The following ENVs are available:
+
+```
+- API_BIND=<teos_api_hostname>
+- API_PORT=<teos_api_port>
+- RPC_BIND=<teos_rpc_hostname>
+- RPC_PORT=<teos_rpc_port>
+- BTC_NETWORK=<btc_network>
+- BTC_RPC_CONNECT=<btc_node_hostname>
+- BTC_RPC_PORT=<btc_node_port>
+- BTC_RPC_USER=<btc_rpc_username>
+- BTC_RPC_PASSWORD=<btc_rpc_password>
+- DEBUG=<debug_bool>
+- DEPS_DEBUG=<deps_debug_bool>
+- OVERWRITE_KEY=<overwrite_key_bool>
+- FORCE_UPDATE=<force_update_bool>
+```
+
+2) Config file:
+
+If you'd like to use a `teos.toml` file instead, one option is to edit the Dockerfile with this line right above the ENTRYPOINT:
+
+	COPY teos.toml /home/teos/.teos
+
+Then build it:
+
+	docker build -f docker/Dockerfile -t teos .
+
+Then run:
+
+	docker run -it teos
+
+### Volume persistence
+
+You may also want to run docker with a volume, so you can have data persistence in `teosd` databases and keys.
+If so, run:
+
+    docker volume create teos-data
+    
+And add the the mount parameter to `docker run`:
+
+    --mount source=teos-data,target=/home/teos/.teos
+
+If you are running `teosd` and `bitcoind` in the same machine, continue reading for how to create the container based on your OS.
+
+### `bitcoind` running on the same machine (UNIX)
+The easiest way to run both together in the same machine using UNIX is to set the container to use the host network.
+	
+For example, if both `teosd` and `bitcoind` are running on default settings, run:
+    
+```
+docker run \
+  --network=host
+  --name teos \
+  --mount source=teos-data,target=/home/teos/.teos \
+  -e BTC_RPC_CONNECT=host.docker.internal \
```

Comment:
I don't think this should be needed for Linux, given we are using --nerwork=host. Can you confirm?

Reply:
Yes you're correct 👍🏻